plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPrompter.java

```diff
@@ -59,13 +59,10 @@ public static CloudSdk getCloudSdk(Shell shell) {
    * @return the Cloud SDK, or {@code null} if unspecified
    */
   public static CloudSdk getCloudSdk(IShellProvider shellProvider) {
-    CloudSdkProvider cloudSdkProvider = new CloudSdkProvider(null);
-    CloudSdk sdk = cloudSdkProvider.getCloudSdk();
-    if (sdk != null) {
-      return sdk;
-    }
-    if (promptForSdk(shellProvider)) {
-      return cloudSdkProvider.getCloudSdk();
+    CloudSdk cloudSdk = new CloudSdkProvider().getCloudSdk();
+    
+    if (cloudSdk != null || promptForSdk(shellProvider)) {
+      return cloudSdk;
```

Comment:
This isn't right: if `cloudSdk == null` (no SDK found), we prompt the user and if they configure a location we return `cloudSdk` which is `null`.  We instead need to re-resolve the SDK.